### DISCUSSION TITLE
How can i use RESTful API in mishard?

### DISCUSSION THREAD

**Discussion by rky0930** [milvus-io/milvus#4860] [Q&A and General discussion] [answered]
How can i use RESTAPI in mishard? 
When deployed on k8s, it seems that only the engine port(ex, 19530) is open, and the web port(ex, 19121) does not exist.

**Comment by yhmo** [comment, marked as answer]
Mishards doesn't support RESTAPI.
Mishards server is written by python, it is a grpc server.

**Reply by rky0930** [reply]
Thanks for answering. Can I ask additional questions?
If I want to call from Ruby on rails server, what options are there? Can I generate Ruby GRPC code myself? or Make a relay server with the given SDK?

**Reply by yhmo** [reply]
Officially we provide python/go/c++/java/restful sdk for users. For Ruby, one way is call restful api from ruby, the other way is create a grpc client by ruby. 
Milvus grpc proto files: /milvus/core/src/grpc/milvus.proto and /milvus/core/src/grpc/status.proto
Milvus restful api document: https://github.com/milvus-io/milvus/blob/1.0/core/src/server/web_impl/README.md

**Reply by rky0930** [reply]
I am going to use mishard, so I can't use RESTAPI. I should consider using GRPC. 
Thanks for the reply !
